vavr/src/main/java/io/vavr/control/Validation.java

```diff
@@ -534,43 +543,45 @@ default void forEach(Consumer<? super T> action) {
     }
 
     /**
-     * Applies a function f to the error of this Validation if this is an Invalid. Otherwise does nothing
+     * Applies a function f to the errors of this Validation if this is an Invalid. Otherwise does nothing
      * if this is a Valid.
      *
-     * @param <U> type of the error resulting from the mapping
-     * @param f   a function that maps the error in this Invalid
+     * @param <U> type of the errors resulting from the mapping
+     * @param f   a function that maps the errors in this Invalid
      * @return an instance of Validation&lt;U,T&gt;
      * @throws NullPointerException if mapping operation f is null
      */
-    default <U> Validation<U, T> mapError(Function<? super E, ? extends U> f) {
+    default <U> Validation<U, T> mapError(Function<Seq<? extends E>, ? extends U> f) {
         Objects.requireNonNull(f, "f is null");
         if (isInvalid()) {
-            final E error = this.getError();
-            return Validation.invalid(f.apply(error));
+            final Seq<? extends E> errors = this.getErrors();
+            return Validation.invalid(f.apply(errors));
         } else {
             return Validation.valid(this.get());
         }
     }
 
-    default <U> Validation<Seq<E>, U> ap(Validation<Seq<E>, ? extends Function<? super T, ? extends U>> validation) {
+    default <U> Validation<E, U> ap(Validation<E, ? extends Function<? super T, ? extends U>> validation) {
         Objects.requireNonNull(validation, "validation is null");
         if (isValid()) {
             if (validation.isValid()) {
                 final Function<? super T, ? extends U> f = validation.get();
                 final U u = f.apply(this.get());
                 return valid(u);
             } else {
-                final Seq<E> errors = validation.getError();
+                final Seq<? extends E> errors = validation.getErrors();
                 return invalid(errors);
             }
         } else {
             if (validation.isValid()) {
-                final E error = this.getError();
-                return invalid(List.of(error));
+                final Seq<? extends E> errors = this.getErrors();
+                return invalid(errors);
             } else {
-                final Seq<E> errors = validation.getError();
-                final E error = this.getError();
-                return invalid(errors.append(error));
+                final Seq<? extends E> errors1 = validation.getErrors();
+                final Seq<? extends E> errors2 = this.getErrors();
+                Seq<E> specificErrors = List.ofAll(errors1);
+                Seq<E> finalErrors = specificErrors.appendAll(errors2);
```

Comment:
specificErrors and finalErrors can be declared `final`. (This is some sort of Vavr convention :) )